lib/models/task.js

```diff
@@ -6,6 +6,17 @@ class Task extends CoreObject {
   run(/*options*/) {
     throw new Error('Task needs to have run() defined.');
   }
+
+  /**
+   * Interrupt comamd with an exit code
+   * Called when the process is interrupted from outside, e.g. CTRL+C or `process.kill()`
+   *
+   * @private
+   * @method onInterrupt
+   */
+  onInterrupt() {
+    process.exit(1);
```

Comment:
having nested exists seems problematic, it would likely be better if we just rejected a given commands run task. Then allowing the primary exit path handle it.